src/components/forms/ComboBox/ComboBox.test.tsx

```diff
@@ -0,0 +1,1111 @@
+    })
+
+    it('does not close menu when an option is selected and the first option is focused and up arrow is pressed', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          defaultValue="apricot"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      // Apple is the item at top of list
+      fireEvent.focus(getByTestId('combo-box-option-apple'))
+      fireEvent.keyDown(getByTestId('combo-box-option-apple'), {
+        key: 'ArrowUp',
+      })
+
+      expect(getByTestId('combo-box-option-apple')).toHaveFocus()
+      expect(getByTestId('combo-box-option-list')).toBeVisible()
+    })
+
+    it('deselects option when pressing delete inside input', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          defaultValue={'avocado'}
+          onChange={jest.fn()}
+        />
+      )
+
+      const input = getByTestId('combo-box-input')
+      userEvent.type(input, '{backspace}')
+
+      expect(getByTestId('combo-box-clear-button')).not.toBeVisible()
+      expect(getByTestId('combo-box-option-list').children.length).toEqual(1)
+    })
+
+    it('focuses the input when an option is focused and shift-tab is pressed', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+      const input = getByTestId('combo-box-input')
+
+      userEvent.click(getByTestId('combo-box-input'))
+      fireEvent.focus(getByTestId('combo-box-option-apricot'))
+      fireEvent.keyDown(getByTestId('combo-box-option-apricot'), {
+        key: 'tab',
+        keyCode: 9,
+        shiftKey: true,
+      })
+
+      expect(input).toHaveFocus()
+    })
+  })
+
+  describe('mouse actions', () => {
+    it('displays options list when input is clicked', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      fireEvent.click(getByTestId('combo-box-input'))
+
+      expect(getByTestId('combo-box-input')).toHaveAttribute(
+        'aria-expanded',
+        'true'
+      )
+      expect(getByTestId('combo-box-option-list')).toBeVisible()
+      expect(getByTestId('combo-box-option-list').childElementCount).toEqual(
+        fruitOptions.length
+      )
+    })
+
+    it('displays options list when input is clicked twice', () => {
+      const { getByTestId } = render(
+        <ComboBox
+          id="favorite-fruit"
+          name="favorite-fruit"
+          options={fruitOptions}
+          onChange={jest.fn()}
+        />
+      )
+
+      fireEvent.click(getByTestId('combo-box-input'))
```

Comment:
should this use the [dblclick event](https://developer.mozilla.org/en-US/docs/Web/API/Element/dblclick_event)?